### ISSUE TITLE
Returning a question redirects to different intent

### ISSUE THREAD

**Issue by ThaWeatherman** [johnwheeler/flask-ask#155]
Since the Dialog interface isn't yet available, I'm home rolling my own with a whole bunch of `if`s. The intent in question requires 4 inputs. I have cases in there for getting each one and saving the rest off to the session while gathering. You can see this intent [here](https://github.com/ThaWeatherman/alexa_skills/blob/master/infant/app.py#L157). There is similar intent with 5 inputs [here](https://github.com/ThaWeatherman/alexa_skills/blob/master/infant/app.py#L224).

I'm testing by trying to go sequentially: `Alexa, ask infant tracker to add a diaper` -> `What is the child's name?` as expected. But then when I give the name, I get a request to the `AddChildIntent` intent [here](https://github.com/ThaWeatherman/alexa_skills/blob/master/infant/app.py#L118) instead of being sent back to the `AddDiaperIntent`. Why would this happen?

**Comment by voutilad** [comment]
From my experience, having very short utterances where it's mostly a slot value can easily lead to this situation. If you're going to do short variable (slot-based) utterances as responses to questions, you'll need an intent designed specifically for receiving the value and then using a session attribute to decide what logic to trigger.

In your case you might need to make [`AddChildIntent  add {Name}`](https://github.com/ThaWeatherman/alexa_skills/blob/master/infant/speech_assets/SampleUtterances.txt#L14) a separate intent related to a prompt for a name that might be used across other states of the app. (The other `AddChildIntent` utterances are probably ok since they include more words than just a slot.)

I ran into this design issue previously when prompting a user for selecting a particular podcast program to play. My case was simpler, so the "select" intent just called the function dealing with handling the "play" intent. The "select" intent was an utterance defined to be match when a user spoke just the program name. (Technically, at any time for any reason, they could just say the name of a correct program and it would start playing it as a result, but this would allow us the spot to put the check for session attributes to see the state of the dialog if needed.)

Just an aside: Flask-Ask has really nothing to do with the intent chosen by Amazon, if people are wondering.

**Comment by ThaWeatherman** [comment]
Yeah I figured this was on Amazon's end. I'm guessing the new Dialog interface would solve this problem.

So basically I need to split all of the dialogs out into separate intents? Then based on a session variable I route it to the proper intent?
